pkg/handler/v1/provider/openstack.go

```diff
@@ -324,29 +330,20 @@ type OpenstackTenantReq struct {
 
 func DecodeOpenstackTenantReq(_ context.Context, r *http.Request) (interface{}, error) {
 	var req OpenstackTenantReq
-
 	req.Username = r.Header.Get("Username")
 	req.Password = r.Header.Get("Password")
 	req.Domain = r.Header.Get("Domain")
 	req.DatacenterName = r.Header.Get("DatacenterName")
 	req.Credential = r.Header.Get("Credential")
-
 	return req, nil
 }
 
-func getOpenstackCredentials(userInfo *provider.UserInfo, credentialName, username, password, domain, tenant, tenantID string, presetProvider provider.PresetProvider) (string, string, string, string, string, error) {
-	if len(credentialName) > 0 {
-		preset, err := presetProvider.GetPreset(userInfo, credentialName)
-		if err != nil {
-			return "", "", "", "", "", fmt.Errorf("can not get preset %s for the user %s", credentialName, userInfo.Email)
-		}
+func getPresetCredentials(userInfo *provider.UserInfo, presetName string, presetProvider provider.PresetProvider) *v1.Openstack {
+	preset, err := presetProvider.GetPreset(userInfo, presetName)
```

Comment:
Please don't swallow the error here, we should distinguish the cases where the credentials are not defined for openstack from the case where an error occurs when fetching the preset at least for logging purposes. I would rather do the following (if you think it's worthy to define a function for this):

```
func getPresetCredentials(userInfo *provider.UserInfo, presetName string, presetProvider provider.PresetProvider) (*credentials, error) {
    p, err := presetProvider.GetPreset(userInfo, presetName)
    if err != nil {
        return nil, fmt.Errorf("can not get preset %s for the user %s", presetName, userInfo.Email)
    }
    if p.Spec.Openstack == nil {
        return nil, fmt.Errorf("credentials for OpenStack provider not present in preset %s for the user %s", presetName, userInfo.Email)
    }
    return &credentials{
			username: p.Spec.Openstack.Username,
			password: p.Spec.Openstack.Password,
			domain:   p.Spec.Openstack.Domain,
			tenant:   p.Spec.Openstack.Tenant,
			tenantID: p.Spec.Openstack.TenantID,
		}, nil
}
```